Sources/StreamChat/Query/FilterDecoding_Tests.swift

```diff
@@ -0,0 +1,221 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+@testable import StreamChat
+import XCTest
+
+class FilterDecoding_Tests: XCTestCase {
+    // MARK: - Exception thrown tests
+
+    func testFilterDecodingThrowExceptionOnEmpty() {
+        checkDecodingFilterThrowException(json: "")
+    }
+
+    func testFilterDecodingThrowExceptionOnInvalidJSON() {
+        checkDecodingFilterThrowException(json: "12345")
+    }
+
+    func testFilterDecodingThrowExceptionOnInvalidKeylessJSON() {
+        checkDecodingFilterThrowException(json: #"{"value"}"#)
+    }
+
+    func testFilterRightSideDecodingThrowExceptionWithMoreThanOneKey() {
+        let filterJson = #"{ "test_key":{ "$eq":"test_value_1", "$ne":"test_value_2" } }"#
+        checkDecodingFilterThrowException(json: filterJson)
+    }
+
+    func testFilterRightSideDecodingThrowExceptionWithoutOperationKey() {
+        let filterJson = #"{ "test_key": { "eq":"test_value_1" } }"#
+
+        checkDecodingFilterThrowException(json: filterJson)
+    }
+
+    func testFilterRightSideDecodingThrowExceptionOnNonExistingOperation() {
+        let filterJson = #"{ "test_key": { "$myspecialOperation":"test_value_1" } }"#
```

Comment:
Let's be 100% sure that it's impossible for the filter to end up encoded this way (ie. if we have something like .customOperator("myspecialOperation") we are in trouble)

Reply:
Yes, for sure, but for decoder, we should test this corner case to be thrown exception and not crash